src/nomad_material_processing/general/schema.py

```diff
@@ -0,0 +1,281 @@
+        description=""" The recipe used for the process. If a recipe is found,
+           all the data is copied from the Recipe within the Process.
+           """,
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.ReferenceEditQuantity,
+        ),
+    )
+    steps = SubSection(
+        description="""
+        The steps of the etching process.
+        """,
+        section_def=EtchingStep,
+        repeats=True,
+    )
+
+
+class EtchingRecipe(Etching, Recipe, EntryData):
+    """
+    A Recipe for an etching process.
+    """
+
+    m_def = Section(
+        a_eln={'hide': ['datetime', 'samples']},
+    )
+
+
+class AnnealingStep(ActivityStep):
+    """
+    A step of annealing process.
+    """
+
+    m_def = Section()
+    duration = Quantity(
+        type=np.float64,
+        description='The elapsed time since the annealing process started.',
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.NumberEditQuantity, defaultDisplayUnit='minute'
+        ),
+        unit='second',
+    )
+    temperature = Quantity(
+        type=np.float64,
+        description='The temperature of the etching process.',
+        a_eln={'component': 'NumberEditQuantity', 'defaultDisplayUnit': 'celsius'},
+        unit='celsius',
+    )
+    agitation = Quantity(
+        type=MEnum(
+            'Magnetic Stirring',
+            'Sonication',
+        ),
+        description='The agitation method used during the etching process.',
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.EnumEditQuantity,
+        ),
+    )
+    annealing_reagents = SubSection(
+        section_def=CompositeSystemReference,
+    )
+
+
+class Annealing(Process, EntryData):
+    """
+    Heat treatment process used to alter the material's properties,
+    such as reducing defects, improving crystallinity, or relieving internal stresses.
+    """
+
+    m_def = Section()
+    recipe = Quantity(
+        type=Reference(SectionProxy('AnnealingRecipe')),
+        description=""" The recipe used for the process. If a recipe is found,
+           all the data is copied from the Recipe within the Process.
+           """,
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.ReferenceEditQuantity,
+        ),
+    )
+    duration = Quantity(
+        type=np.float64,
+        description='The elapsed time since the annealing process started.',
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.NumberEditQuantity, defaultDisplayUnit='minute'
+        ),
+        unit='second',
+    )
+    steps = SubSection(
+        description="""
+        The steps of the annealing process.
+        """,
+        section_def=AnnealingStep,
+        repeats=True,
+    )
+
+
+class AnnealingRecipe(Annealing, Recipe, EntryData):
+    """
+    A Recipe for an annealing process.
+    """
+
+    m_def = Section(
+        a_eln={'hide': ['datetime', 'samples']},
+    )
+
+
+class CleaningStep(ActivityStep):
```

Comment:
We should use `ProcessStep` for these I guess:
```suggestion
class CleaningStep(ProcessStep):
```

Reply:
solved